pkg/operator/ceph/object/controller.go

```diff
@@ -525,43 +506,40 @@ func (r *ReconcileCephObjectStore) reconcileMultisiteCRs(cephObjectStore *cephv1
 	return cephObjectStore.Name, cephObjectStore.Name, cephObjectStore.Name, nil, reconcile.Result{}, nil
 }
 
-func monitoringChannelKey(o *cephv1.CephObjectStore) string {
-	return types.NamespacedName{Namespace: o.Namespace, Name: o.Name}.String()
-}
-
-func (r *ReconcileCephObjectStore) startMonitoring(objectstore *cephv1.CephObjectStore, objContext *Context, namespacedName types.NamespacedName) error {
-	channelKey := monitoringChannelKey(objectstore)
-
-	// Start monitoring object store
-	if r.objectStoreContexts[channelKey].started {
-		logger.Info("external rgw endpoint monitoring go routine already running!")
+// handle upgrade from v1.10; we need to remove the health checker bucket from the store, or
+// deleting the store will be stuck on dependents because an unknown bucket exists
+// TODO: remove this for Rook v1.12 release
+var removeDeprecatedHealthCheckBucket = func(ctx context.Context, opsCtx *AdminOpsContext, cos *cephv1.CephObjectStore) error {
+	healthCheckBucket := genHealthCheckerBucketName(string(cos.UID))
+	doPurge := true // purge all content of the bucket
+	bucket := admin.Bucket{Bucket: healthCheckBucket, PurgeObject: &doPurge}
+	remErr := opsCtx.AdminOpsClient.RemoveBucket(ctx, bucket)
+
+	if remErr == nil {
+		logger.Info("successfully deleted deprecated health checker bucket")
 		return nil
 	}
 
-	rgwChecker, err := newBucketChecker(r.context, objContext, r.client, namespacedName, &objectstore.Spec)
-	if err != nil {
-		return errors.Wrapf(err, "failed to start rgw health checker for CephObjectStore %q, will re-reconcile", namespacedName.String())
+	if errors.Is(remErr, admin.ErrNoSuchBucket) {
+		logger.Debug("deprecated health checker bucket already does not exist: NoSuchBucket")
+		return nil
 	}
 
-	logger.Infof("starting rgw health checker for CephObjectStore %q", namespacedName.String())
-	go rgwChecker.checkObjectStore(r.objectStoreContexts, channelKey)
-
-	// Set the monitoring flag so we don't start more than one go routine
-	r.objectStoreContexts[channelKey].started = true
-
-	return nil
-}
-
-// cancel monitoring. This is a noop if monitoring is not running.
-func (r *ReconcileCephObjectStore) stopMonitoring(objectstore *cephv1.CephObjectStore) {
-	channelKey := monitoringChannelKey(objectstore)
-
-	_, monitoringContextExists := r.objectStoreContexts[channelKey]
-	if monitoringContextExists {
-		// stop the monitoring routine
-		r.objectStoreContexts[channelKey].internalCancel()
-
-		// remove the monitoring routine from the map
-		delete(r.objectStoreContexts, channelKey)
+	if errors.Is(remErr, admin.ErrNoSuchKey) {
```

Comment:
Why do we need to go the extra mile to log if the bucket exists? Seems like this can just be a best effort to remove the bucket, and we don't need lines 528-542.

Reply:
We can't just do a best-effort to remove the bucket. It actually has to be gone in order to not cause potential issues with bucket checking during deletion.